### DISCUSSION TITLE
v0.17.0

### DISCUSSION THREAD

**Discussion by JohnnyMorganz** [JohnnyMorganz/StyLua#660] [General]
### Added

- Added support for "sort requires", which sorts top-level statements of the form `local NAME = require(EXPR)` lexicographically on `NAME`.
  We do this by treating a group of consecutive requires as a "block", and then sorting **only within** the block. Any other statement, or an empty line, between require statements will split the group into two separate blocks (and can be used to separate the sorting). A block of requires will not move around the file.
  Roblox Luau statements of the form `local NAME = game:GetService(EXPR)` will also be sorted separately.

This feature is disabled by default. To enable it, add the following to your `stylua.toml`:

```toml
[sort_requires]
enabled = true
```

Note: we assume that all requires are **pure** with no side effects. It is not recommended to use this feature if the ordering of your requires matter.

- Added support for [EditorConfig](https://editorconfig.org/), which is taken into account only if no `stylua.toml` was found.

This feature is enabled by default, it can be disabled using `--no-editorconfig`.

- Published StyLua to the [Docker Hub](https://hub.docker.com/r/johnnymorganz/stylua)

---

## What's Changed
* Sort requires by @JohnnyMorganz in https://github.com/JohnnyMorganz/StyLua/pull/653
* Add default `editorconfig` feature by @LEI in https://github.com/JohnnyMorganz/StyLua/pull/645
* extension: Pass cwd as workspace folder for version check by @JohnnyMorganz in https://github.com/JohnnyMorganz/StyLua/pull/659
* Add Dockerfile by @eberkund in https://github.com/JohnnyMorganz/StyLua/pull/655

## New Contributors
* @LEI made their first contribution in https://github.com/JohnnyMorganz/StyLua/pull/645
* @eberkund made their first contribution in https://github.com/JohnnyMorganz/StyLua/pull/655

**Full Changelog**: https://github.com/JohnnyMorganz/StyLua/compare/v0.16.1...v0.17.0

<hr /><em>This discussion was created from the release <a href='https://github.com/JohnnyMorganz/StyLua/releases/tag/v0.17.0'>v0.17.0</a>.</em>

**Comment by hinell** [comment]
Can we have `.editorcong` and `.stylua.toml` merged if both are present in the worktree? Thanks. It would reduce amount of duplicate config alot.

**Reply by JohnnyMorganz** [reply]
This is a bit more challenging to implement - and ideally we enforce a single source of truth for config.

I'm pretty sure .editorconfig supports all the config that stylua.toml does, so you don't necessarily need both files
